File: skyspaceeleven_ui/src/services/ScheduledEventService.js
import http from "../http-common"; 

class ScheduledEventService {
  getAllScheduledEvents(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/scheduledEvent/scheduledEvents`, searchDTO);
  }

  get(scheduledEventId) {
    return this.getRequest(`/scheduledEvent/${scheduledEventId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/scheduledEvent?field=${matchData}`, null);
  }

  addScheduledEvent(data) {
    return http.post("/scheduledEvent/addScheduledEvent", data);
  }

  update(data) {
  	return http.post("/scheduledEvent/updateScheduledEvent", data);
  }
  
  uploadImage(data,scheduledEventId) {
  	return http.postForm("/scheduledEvent/uploadImage/"+scheduledEventId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ScheduledEventService();

File: src/main/java/com/skyspaceeleven/service/ScheduledEventService.java
package com.skyspaceeleven.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspaceeleven.domain.ScheduledEvent;
import com.skyspaceeleven.dto.ScheduledEventDTO;
import com.skyspaceeleven.dto.ScheduledEventSearchDTO;
import com.skyspaceeleven.dto.ScheduledEventPageDTO;
import com.skyspaceeleven.dto.ScheduledEventConvertCriteriaDTO;
import com.skyspaceeleven.service.GenericService;
import com.skyspaceeleven.dto.common.RequestDTO;
import com.skyspaceeleven.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ScheduledEventService extends GenericService<ScheduledEvent, Integer> {

	List<ScheduledEvent> findAll();

	ResultDTO addScheduledEvent(ScheduledEventDTO scheduledEventDTO, RequestDTO requestDTO);

	ResultDTO updateScheduledEvent(ScheduledEventDTO scheduledEventDTO, RequestDTO requestDTO);

    Page<ScheduledEvent> getAllScheduledEvents(Pageable pageable);

    Page<ScheduledEvent> getAllScheduledEvents(Specification<ScheduledEvent> spec, Pageable pageable);

	ResponseEntity<ScheduledEventPageDTO> getScheduledEvents(ScheduledEventSearchDTO scheduledEventSearchDTO);
	
	List<ScheduledEventDTO> convertScheduledEventsToScheduledEventDTOs(List<ScheduledEvent> scheduledEvents, ScheduledEventConvertCriteriaDTO convertCriteria);

	ScheduledEventDTO getScheduledEventDTOById(Integer scheduledEventId);







}





